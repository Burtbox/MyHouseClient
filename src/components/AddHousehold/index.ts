import AddHousehold, { AddHousehold as DisconnectedAddHousehold } from './AddHousehold';

export { DisconnectedAddHousehold };
export default AddHousehold;
