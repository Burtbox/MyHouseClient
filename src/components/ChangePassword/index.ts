import ChangePassword, { ChangePassword as DisconnectedChangePassword } from './ChangePassword';

export default ChangePassword;

export { DisconnectedChangePassword };
