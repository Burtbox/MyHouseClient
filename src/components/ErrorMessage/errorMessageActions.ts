import { IErrorMessageAction } from './errorMessageInterfaces';

// Export Actions
export enum errorMessageActions {
    ADD_ERROR = 'ADD_ERROR',
    REMOVE_ERROR = 'REMOVE_ERROR',
}

export function addError(errorMessageText: string) {
    const response: IErrorMessageAction = {
        errorMessageText,
        type: errorMessageActions.ADD_ERROR,
    };
    return response;
}

export function removeError() {
    const response: IErrorMessageAction = {
        type: errorMessageActions.REMOVE_ERROR,
        errorMessageText: undefined,
    };
    return response;
}
