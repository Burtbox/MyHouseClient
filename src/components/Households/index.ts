import Households, { Households as DisconnectedHouseholds } from './Households';

export default Households;

export { DisconnectedHouseholds };
