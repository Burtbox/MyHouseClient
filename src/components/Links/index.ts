import Links, { Links as DisconnectedLinks } from './Links';

export default Links;
export { DisconnectedLinks };
