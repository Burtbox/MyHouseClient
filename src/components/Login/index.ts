import Login, { Login as DisconnectedLogin } from './Login';

export default Login;

export { DisconnectedLogin };
