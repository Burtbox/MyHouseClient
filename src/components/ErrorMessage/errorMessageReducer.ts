import { errorMessageActions } from './errorMessageActions';
import { IErrorMessageState, IErrorMessageAction } from './errorMessageInterfaces';

function errorMessageReducer(
    state: IErrorMessageState = { errorMessageText: null },
    action: IErrorMessageAction,
) {
    switch (action.type) {
    case errorMessageActions.ADD_ERROR:
        return {
            ...state,
            errorMessageText: action.errorMessageText,
        };
    case errorMessageActions.REMOVE_ERROR:
        return {
            ...state,
            errorMessageText: null,
        };
    default:
        return state;
    }
}

// Export Reducer
export default errorMessageReducer;
