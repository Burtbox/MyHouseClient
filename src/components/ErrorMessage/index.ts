import ErrorMessage, { ErrorMessage as DisconnectedErrorMessage } from './ErrorMessage';

export default ErrorMessage;

export { DisconnectedErrorMessage };
