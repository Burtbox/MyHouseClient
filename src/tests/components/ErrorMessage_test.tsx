import * as React from 'react';
import { shallow } from 'enzyme';
import { DisconnectedErrorMessage } from '../../components/ErrorMessage';
import { IErrorMessageProps } from '../../components/ErrorMessage/errorMessageInterfaces';

describe('ErrorMessage test suite', () => {
    test('Render ErrorMessage component', () => {
        const props: IErrorMessageProps = {
            dispatch: undefined,
            errorMessageText: 'unit test error',
        };
        const ErrorMessageScreen = shallow(
            <DisconnectedErrorMessage {...props} />,
        );
        expect(ErrorMessageScreen).toMatchSnapshot();
    });
});
