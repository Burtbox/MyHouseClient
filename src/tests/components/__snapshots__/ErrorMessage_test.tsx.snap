// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage test suite Render ErrorMessage component 1`] = `
<div>
  <div />
</div>
`;
